### PR TITLE
docs(prompts): twenty-one help-wanted starter kits, an index, and the unit-test file that stopped compiling

### PR DESCRIPTION
**Pull request by joyful-ii-V-I**
## What this is

Twenty-one open problems in ripwire, each turned into a starter kit: a self-contained prompt under
`prompts/help-wanted/` with the research already done, so someone new to the codebase can take one
on without first reverse-engineering where the problem lives.

Each prompt carries the same things: why the problem matters, with the measurement rather than an
adjective; file:line pointers into the code, each with its symbol; a reproduction with recorded
output; the design space, including what has already been tried and rejected; the non-negotiables
from `CONTRIBUTING.md` and `docs/METHODOLOGY.md`; the acceptance criteria; the traps met while the
kit was written; and what the eventual pull request's description must contain.

`prompts/help-wanted/README.md` indexes the twenty prompts in that directory by difficulty, and
names `prompts/full-audit.md` as the twenty-first kit. That top-level prompt is also rewritten here,
to fold in what the 2026-09-10 audit learned — most importantly that it was fooled three times by
its own instruments, so every lens now has a control to run first.

This is documentation plus four gate files and one test-file fix. Nothing under `src/` changes.

## How a contributor picks one up

1. **Comment on the kit's issue to claim it.** Each kit has its own issue, labelled `help wanted`
   and titled with the slug. The claim exists so two people do not build the same thing.
2. **Paste the prompt into a coding agent at the root of a checkout.** Build first — a plain dev
   build, never `-DCMAKE_BUILD_TYPE=Release`.
3. **The prompt writes a plan and stops.** Every prompt in `prompts/` ends this way, and these are no
   exception. The first deliverable is the plan, not code: which files change, which arms flip, what
   gets measured, what deliberately stays the same.
4. **A maintainer agrees the plan.** Read it, cut what you disagree with, then say go. Several kits
   suggest posting the plan — or just the STEP 0 measurement — as a draft pull request first, which
   is the cheapest place to find out that the approach is wrong.

Three kits are marked `good first issue`: `struct-layout-doctor`, `next-uses-bare-name` and
`uses-qualified-selector`. Each has a root cause that fits in one function and an acceptance test
that already exists.

## What the known-gap arms are for

Four kits ship their acceptance test up front, as arms in existing gates:

| Gate | Kit | KNOWN GAP arms |
| --- | --- | --- |
| `test/usesselectorcheck.sh` | `uses-qualified-selector` | 12 |
| `test/stdqualcheck.sh` §11 | `cpp-nested-std-namespaces` | 7 |
| `test/fieldnarrowcheck.sh` | `ts-literal-receivers` | 5 |
| `test/yamllangcheck.sh` | `nesting-refusals-visible` | 4 |
| `test/mdsectioncheck.sh` | `nesting-refusals-visible` | 4 |

These arms **pin today's wrong answer**, and they pass today. That is the point: the gap is written
down as an executable assertion rather than a paragraph, each arm prints
`KNOWN GAP (help wanted: prompts/help-wanted/<slug>.md)` so nobody mistakes it for intended
behaviour, and each carries its FIXED expectation in a comment beside it. **Flipping an arm from the
wrong literal to the right one is the finish line** — a contributor knows exactly when the work is
done, and a reviewer can see it in the diff.

Each block also carries controls that any fix must keep green — true edges, precision controls that
catch a too-broad fix, and presence guards so a block cannot pass vacuously. Every block was
observed red against a wrapper faking the naive fix before it was committed.

The four gates are existing files. No gate file is added, so `test/regression.sh` and the generated
gate count do not move.

## The `verify_pagerank.cpp` fix, called out separately

One commit here is not documentation: **`36e80401`, a one-line fix to `test/verify_pagerank.cpp`.**

`2bbd134d` (2026-08-17) gave `pageRankDouble` a `PageRankRun { iterationCount, hasConverged }`
return so a caller could tell a converged run from an exhausted one. The empty-graph boundary case
in `test/verify_pagerank.cpp` still assigned that struct to a `const unsigned`, so **the file
stopped compiling that day and nobody noticed for three and a half weeks** — the unit-test targets
sit behind `option(RIPWIRE_TESTS)`, which is OFF by default, and no CI job turns it on.

That is exactly the gap the `graph-unit-tests` kit asks a contributor to close, so the file has to
build before the kit is worth handing to anyone. The fix reads the member the case is actually
about:

```
const unsigned iterationCount = rw::pageRankDouble( csr, {}, {}, rank, preciseConfig ).iterationCount;
```

`hasConverged` is deliberately left alone — the kit has to decide what convergence means on an empty
graph before asserting anything about it. Wiring these targets into CI is the kit's job, not this
pull request's.

Verified with the option on, in a build tree outside the working tree so the crawl is unperturbed:

```
cmake -S . -B <tree> -DRIPWIRE_TESTS=ON
cmake --build <tree> --target ripwire_test_pagerank ripwire_test_csr -j 4
<tree>/ripwire_test_pagerank   ->  1 test case,  16 assertions, SUCCESS
<tree>/ripwire_test_csr        ->  2 test cases, 23 assertions, SUCCESS
ctest -R 'ripwire.pagerank|ripwire.csr'   ->  2/2 passed
```

## Gates run

Full suite, in the foreground on a plain dev build:

```
python3 test/pargates.py . ./build/ripwire -j 4
gates=616  pass=613  skip=3  fail=0  tree_writes=0   ALL PASS
```

The three skips are environmental and expected: `argvdiffcheck` and `editchecknotecheck` want a
reference binary that is not set, and `g1freshcheck` wants an `asan/` tree that is not configured.

Each known-gap gate also run individually, in the foreground:

```
usesselectorcheck   57 PASS, 0 FAIL   (12 KNOWN GAP)   ALL PASS
stdqualcheck        50 PASS, 0 FAIL   ( 7 KNOWN GAP)   ALL PASS
fieldnarrowcheck    30 PASS, 0 FAIL   ( 5 KNOWN GAP)   ALL PASS
mdsectioncheck      98 PASS, 0 FAIL   ( 4 KNOWN GAP)   ALL PASS
yamllangcheck       51 PASS, 0 FAIL   ( 4 KNOWN GAP)   ALL PASS
```

Generated artifacts and quality:

```
python3 docs/gatecount_build.py --check --root .
  -> 8 marked site(s) in 3 file(s) all state the loop in test/regression.sh names 602
python3 docs/limits_build.py --check
  -> docs/LIMITS.md matches src/ (208 caps, 7 parameters, 111 classified)
ripwire . --quality-delta=origin/main..HEAD
  -> regressions="0" gating="0"   (exit 0)
```

The gate count is unchanged at 602, because every known-gap arm extends a gate file that already
exists.

`README.md` is untouched. `test/readmedriftcheck.sh` (I1) counts only top-level `prompts/*.md`, and
the kits live in `prompts/help-wanted/`, so the "twelve self-contained orchestrator prompts"
sentence stays correct.

## A note on the index

`prompts/help-wanted/README.md` was written when each kit was going to arrive on its own pull
request. It does not — they land together here, and each problem gets an issue instead. The last
commit fixes the three places that said otherwise, and drops a `PR` column whose every row held a
`#TBD-<slug>` placeholder. A placeholder for a number that does not exist yet is a reference waiting
to be wrong; the issue is found by the slug the first column already gives.

The table was otherwise already correct against this branch: its twenty rows are exactly the twenty
prompts under `prompts/help-wanted/`, with `full-audit` named in the prose as the twenty-first, and
no row for either kit that was retired before assembly.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: prompts/full-audit.md [hunk 1]
```diff
@@ -1,99 +1,262 @@
...
+- one very long line;
+- invalid UTF-8;
+- merge-conflict markers;
+- one enormous initializer.
 
-Wrong output, crashes, silent truncation, a flag that does not do what `--help` says. Start where
-the tool itself says the risk is: `./build/ripwire . --hotspots`, `--lint`, `--clones`, and
-`--report`. Then read the code those point at. A bug is only a finding once you have a **reproducer
-argv** that shows it.
+A crash is HIGH. Silent corruption only shows under the sanitizer.
```
[the remark this answers is not in the document]
Comment (joyful-ii-V-I) [reply prompts/full-audit.md]: Valid, and the prompt contradicted itself two paragraphs later. Fixed in 658b2ec6.

The sentence was "A crash is HIGH. Silent corruption only shows under the sanitizer." The very next
section of the same lens — "Every omission counted" — is a conservation oracle for exactly the
corruption a sanitizer cannot see: PR #136's resolve census caught a silent decline that answered
`count=0` on 22% of one corpus's call references, and it raises `DEGRADED_PATH_ALERT` on plain
builds, not under ASan. A wrong count, a dropped row and a mis-sorted list are all in-bounds and
exit 0 under the sanitizer.

The lens now splits the two halves and names the oracle for the second:

> A crash is HIGH. Silent corruption is the other half, and the sanitizer sees only the part of it
> that is a memory-safety violation — the out-of-bounds write, the use-after-free. A wrong count, a
> dropped row, a mis-sorted list stays in bounds and exits 0 under the sanitizer exactly as it does
> in the plain build. Catch those with an independent oracle on the same input — the conservation
> identity below, a second verb that has to agree, a byte-identical re-run — and run it in the plain
> build too.

"the conservation identity below" points at the section that immediately follows, so the lens hands
the reader the oracle rather than only warning about the gap. The rest of the hostile-input list is
unchanged.

File: prompts/help-wanted/next-uses-bare-name.md [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+| `cpp/alpha/alpha.cpp::Alpha::crender` (canonical id) | `--uses=cpp/alpha/alpha.cpp::Alpha::crender` | `count="0"`, **no** `call_sites_of_name=` at all |
+| `Alpha::crender` (`Scope::name`) | `--uses=Alpha::crender` | `count="0"`, **no** `call_sites_of_name=` at all |
+
+Every row above carries `declined_calls="1"`; the declined `crender` site is
+`cpp/caller/caller.cpp:3`. The two `::` spellings are worse because `--uses` matches sites against
+the whole spelling there (STEP 2), and that is true even when nothing was declined:
+`--callers=cpp/pair/one.cpp::One::ctwin` answers `count="1"` with
+`next="--uses=cpp/pair/one.cpp::One::ctwin"`, and that answer is `count="0"`. **That bound case is a
+separate, older gap. Leave its output byte-identical in this change and name it in the PR as a
+follow-up.**
```
[the remark this answers is not in the document]
Comment (joyful-ii-V-I) [reply prompts/help-wanted/next-uses-bare-name.md]: Valid. Fixed in f138dd3a.

Verified both halves. `uses-qualified-selector.md` is marked "no prerequisite" and already carries
the reciprocal note — it names this kit, says the two "touch different code, can land in either
order, and must not be combined into one PR", and tells its own reader to re-run
`test/declinecheck.sh` if this kit landed first. This kit carried no such note, so the asymmetry was
real.

The pinned baseline itself reproduces exactly at the head it names: on `test/declinefix`,
`--callers=cpp/pair/one.cpp::One::ctwin` is `count="1"` with
`next="--uses=cpp/pair/one.cpp::One::ctwin"`, and that pointer answers `count="0"`. That is precisely
the row `uses-qualified-selector.md` turns into `count="1"`, so if it lands first the "leave its
output byte-identical" instruction and the STEP 6 follow-up bullet are both stale. The sibling also
says its change makes the declined `::` spellings start carrying the `file:name` disclosure, which
stales the two table rows reading "**no** `call_sites_of_name=` at all".

Line drift is real too, independent of ordering: `src/graph.h` and `src/graphlegend.h` have already
moved on `main` since the head this kit pins, so its line numbers there are stale today. The kit
already mitigated that ("every pointer also names its symbol"); it had no equivalent for outputs.

What changed:

- The provenance paragraph now states that the baseline is the commit the contributor branches from,
  not the pinned head: build the base, run the STEP 1 block on it, record what it prints, and read
  every "leave this byte-identical" against that. Where the base disagrees with a printed tag, the
  base wins and the plan says which rows moved.
- A **sibling kit** paragraph after the bound `::` case gives the two commands that distinguish the
  landing orders, and says what each order implies for the byte-identity requirement, the two
  `call_sites_of_name=` table rows and the follow-up bullet. It restates that neither kit is a
  prerequisite for the other and that they must not be combined.
- The STEP 6 follow-up bullet is now conditional on the base.

No claim about the declined-call gap itself moved, and nothing under `src/` or `test/` changed.

File: prompts/help-wanted/zig-language.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+- **Types are values.** `const Point = struct { … };` is how a type gets a name, and the same goes for
+  `enum`, `union`, `opaque` and `error{ … }`. Name the symbol after the `const` binding, and give it
+  the kind of the container it binds.
+- **Methods are `fn` declarations inside a container body.** Scope-qualify them: `Point.distance`, not
+  a bare `distance`. Read `src/ingest_elixir.h` first anyway — add-a-language.md calls it the most
+  conventional extraction module — and follow how the existing extractors build scoped symbol ids
+  rather than inventing a new scheme.
+- **Files are values too.** `const fs = @import("fs.zig");` binds that file's top-level declarations
+  to `fs`, so `fs.openFile()` is a call into `fs.zig`. This is the resolver's most valuable edge.
+  Make it the first thing a decoy fixture attacks.
```
[the remark this answers is not in the document]
Comment (joyful-ii-V-I) [reply prompts/help-wanted/zig-language.md]: Valid. Fixed in 0c0b76e3, by making the deferral explicit rather than by making STEP 5 mandatory.

The ambiguity was real and had four sites. `prompts/add-a-language.md` STEP 5 is titled "dependency
edges (optional, and a separate round)", and this kit inherited that — it labelled the `@import`
section "(STEP 5, a separate round)" and its PR-description checklist already said "Deferred work:
`@import` edges if they are not in this PR". Against that, three requirements read as unconditional:
the "Files are values too" bullet called `fs.openFile()` "the resolver's most valuable edge" and said
to attack it first with a decoy; the case-insensitive-filesystem trap said "Plant the decoy"; and
acceptance criterion 4 required a case-collision import fixture. All three are `@import` resolution,
so the kit asked for work it also permitted deferring.

Since the parent prompt makes STEP 5 optional, the fix keeps it optional here and states the scope
both ways:

- A new **Scope** paragraph opens the design notes: the first PR is registration, definitions, calls,
  visibility, scaling and disclosure; `@import` resolution is a second round unless the plan says
  otherwise; and the plan must say which, before anything is written. It names the three requirements
  that hang on the answer.
- The `fs.openFile()` bullet now says it *is* `@import` resolution and follows Scope. Deferred, it
  binds nothing — no edge, and above all no guess at a same-named definition elsewhere in the corpus.
- The dependency-edges section says its four rules apply only if STEP 5 is in the PR; deferred,
  `@import` yields no edge at all, external or otherwise.
- The case-collision decoy is planted either way, and the trap and acceptance criterion 4 each state
  what it proves in each case: with STEP 5, that the specifier resolved through the corpus index and
  not by a case-folded name; without, that nothing bound, pinned as disclosed behaviour rather than
  left as a silent absence.
- Acceptance criterion 8 adds the deferred case to disclosure: "`@import` is not resolved to a file"
  is then a blind spot named in the gate header, the README paragraph and the PR.

Nothing about the grammar pins, the symbol kinds or the parse-rate measurement moved.
